lib/replicator.js

```diff
@@ -1506,21 +1510,21 @@ module.exports = class Replicator {
     this._maybeResolveIfAvailableRanges()
   }
 
-  attachTo (protomux) {
-    const makePeer = this._makePeer.bind(this, protomux)
+  attachTo (protomux, session) {
+    const makePeer = this._makePeer.bind(this, protomux, session)
 
     protomux.pair({ protocol: 'hypercore/alpha', id: this.discoveryKey }, makePeer)
-
     this._ifAvailable++
     protomux.stream.opened.then((opened) => {
       this._ifAvailable--
       if (opened) makePeer()
+      else if (session) session.close().catch(noop)
```

Comment:
Double-checking that this is the right closed condition

Reply:
Yep, the stream was destroyed before channels could be opened -> kill the replication session